lib/views/item-scene.js

```diff
@@ -92,8 +108,29 @@ class ItemScene extends React.Component {
     Linking.openURL(url);
   }
 
-  renderDistance(distance = 0) {
+  renderDistance() {
+    let distance = this.state.distance;
+
+    // No distance and no geographic coordinates
+    if (typeof distance !== 'number' &&
+      (typeof this.props.item.latitude !== 'number' ||
+      typeof this.props.item.longitude !== 'number')) {
+      return null;
+    }
+
+    // mDNS beacon
+    if (distance === -1) {
+      return null;
+    }
+
+    if (typeof distance !== 'number' &&
+      typeof this.props.item.latitude === 'number' &&
+      typeof this.props.item.longitude === 'number') {
+      distance = this.computeDistance(this.props.location.pos.coords);
```

Comment:
<img width="242" alt="screen shot 2016-11-18 at 14 14 59" src="https://cloud.githubusercontent.com/assets/1020551/20448559/7b2c32f6-ad99-11e6-9288-673dc9f7a849.png">

I don't think my device has got a location callback yet, so this is throwing.